python/utils/OpaqueArguments.h

```diff
@@ -185,6 +189,87 @@ inline std::string mlirTypeToString(mlir::Type ty) {
   return msg;
 }
 
+/// @brief Return the size and member variable offsets for the input struct.
+inline std::pair<std::size_t, std::vector<std::size_t>>
+getTargetLayout(func::FuncOp func, cudaq::cc::StructType structTy) {
```

Comment:
We should move this function to a "common" place instead of copying it. The C++ code may need to do this as well in more places than it does at the moment.

Reply:
Note: we should also start using the size of and offset of operations in the IR when appropriate and when offset of is completed.

Reply:
Will do.

Reply:
Actually I think I'll leave this as a follow on as I'm not sure where to put it. I tried to put in the Factory.h/Factory.cpp but I ran into linking issues bc this code requires the CodeGen library.

Reply:
#1951